### ISSUE TITLE
Reset counter did not execute

### ISSUE THREAD

**Issue by DrDBanner** [invoiceninja/ui#1822]
## Setup
- Version: 5.9.4
- Environment: ZIP

## Interface
- Flutter: []
- React: [x]
- Both: []

## Checklist
- Can you replicate the issue on our v5 demo site https://demo.invoiceninja.com or https://react.invoicing.co/demo?
NO

- Have you searched existing issues?
YES

- Have you reported this to Slack/forum before posting?
NO

- Have you inspected the logs in storage/logs/laravel.log for any errors?
NO

## Describe the bug
The automatic counter reset was set to monthly but did not reset as expected. Should have executed on the 1st of june at 0:00:01 I guess.

### Expected Behavior
Reset counter.

## Additional context
No

### Screenshots
<img width="241" alt="Bildschirmfoto 2024-06-10 um 10 51 15" src="https://github.com/invoiceninja/invoiceninja/assets/25568865/ecac7274-aba0-4bff-aa4e-77976afdf0b3">
<img width="1251" alt="Bildschirmfoto 2024-06-10 um 10 50 45" src="https://github.com/invoiceninja/invoiceninja/assets/25568865/3e4b2ec1-e9c0-40ae-b67a-687b74be9adf">
<img width="1261" alt="Bildschirmfoto 2024-06-10 um 10 50 27" src="https://github.com/invoiceninja/invoiceninja/assets/25568865/7c594c5b-a34a-4bcc-ba91-508fea55f552">

**Comment by turbo124** [comment]
So it did reset? You have the 10th of june selected as the date?

**Comment by DrDBanner** [comment]
This field shows my current date. Today it shows the 13th of june. A Date selector appears if I click into the field. When I initially set it to monthly there's a chance I did not put in any date to 'Next Reset'. I can't remember. If so, my expectation was most likely it would determine the date on it's own as long as I do not put any different date. Seems like the documentation doesn't cover what's mandatory here. Wouldn't it be ideal if I change the 'Reset Counter' value the 'Next Reset' date gets determined based on what I choose from on 'Reset Counter'? At least it should be covered when I do not put a date into 'Next Reset' nothing will happen.

**Comment by turbo124** [comment]
The date in the reset counter indicates the next time that the reset date counter will be performed. Once it is triggered. A new date is inserted.

**Comment by DrDBanner** [comment]
Do you want me to do anything specific now?

**Comment by turbo124** [comment]
@DrDBanner If you want the counter to reset on the 1st of July. Set the reset date to there. Let us know if the counter does not reset as expected after that date.

**Comment by DrDBanner** [comment]
In a different installation I have noticed that I set a date there and it was saved. So, I guess in this installation I did not set any date at all. We can argue now who's right. At least I find the userinterface unclear, as well as the documentation in this particular case. So, perhaps adding a one-liner to the documentation, or a one liner to the interface may fix already the issue. Alternatively, making the field mandatory once anything other than 'never' is selected in the interval field.
